File: solutions/5/5.js
function leastCommonMultipleUntil(limit) {
    let lmc = 1;
    let divisor = 2;
    let numbers = Array(limit).fill(0).map((x, i) => i + 1);

    while (1) {

        while (numbers.filter(x => x % divisor == 0).length > 0) {
            numbers = numbers.map(x => x % divisor ? x : x / divisor);

            lmc *= divisor;
        }

        if (Math.max(...numbers) == 1) {
            break;
        }

        divisor += 1;
    }

    return lmc;
}

function main() {
    let result = leastCommonMultipleUntil(20);

    console.log(result);
}

main();

File: solutions/5/5.py
def leastCommonMultipleUntil(limit):
    lmc = 1
    divisor = 2
    numbers = list(range(1, limit + 1))

    while True:
        while any(number % divisor == 0 for number in numbers):
            numbers = [number if number % divisor else number / divisor for number in numbers]
            lmc *= divisor
        if max(numbers) == 1:
            break
        divisor += 1
    return lmc

def main():
    result = leastCommonMultipleUntil(20)
    print(result)

if __name__ == "__main__":
    main()
